editor/project_converter_3_to_4.cpp

```diff
@@ -1888,11 +1893,325 @@ class ProjectConverter3To4::RegExContainer {
 	}
 };
 
+class ProjectConverter3To4::Scene {
+private:
+	HashMap<int, String> content;
+	HashMap<int, EmbeddedScript> scripts;
+
+public:
+	void push_line(const String &line) {
+		content[content.size()] = line;
+	}
+
+	void append_string(const String &str) {
+		// Append sub-string to last line of the scene
+		//
+		// Don't know how it happened but the embedded script
+		// may not have empty last line
+		content[content.size() - 1] += str;
+	}
+
+	int push_script(const EmbeddedScript &script) {
+		int ret = scripts.size();
+		scripts[ret] = script;
+		return ret;
+	}
+
+	void push_script_line(int script_id, const String &line) {
+		scripts[script_id].content.push_back(line);
+	}
+
+	bool has_embedded_scripts() const {
+		return !scripts.is_empty();
+	}
+
+	int get_script_count() const {
+		return scripts.size();
+	}
+
+	Vector<String> get_script_lines(int script_id) const {
+		Vector<String> ret;
+		int max = scripts[script_id].content.size();
+		for (int i = 0; i < max; i++) {
+			ret.push_back(scripts[script_id].content[i].c_unescape());
+		}
+		return ret;
+	}
+
+	Vector<String> get_scene_lines() const {
+		Vector<String> ret;
+		for (uint32_t i = 0; i < content.size(); i++) {
+			ret.append(content[i]);
+		}
+		return ret;
+	}
+};
+
 ProjectConverter3To4::ProjectConverter3To4(int p_maximum_file_size_kb, int p_maximum_line_length) {
 	maximum_file_size = p_maximum_file_size_kb * 1024;
 	maximum_line_length = p_maximum_line_length;
 }
 
+Vector<ProjectConverter3To4::ScriptPart> ProjectConverter3To4::parse_gdscript_line(const String &line) {
+	Vector<ScriptPart> ret;
+
+	if (line.strip_edges().is_empty()) {
+		ScriptPart p;
+		p.content = line;
+		p.exclude = false;
+		ret.push_back(p);
+		return ret;
+	}
+
+	char32_t quote_chr = '"';
+	bool escaped = false;
+	bool literal = false;
+	int start = 0;
+	int pos = 0;
+	int max = line.length();
+
+	for (int i = 0; i < max; i++) {
+		char32_t chr = line.get(i);
+		switch (chr) {
+			case '\\':
+				escaped = !escaped;
+				break;
+			case '"':
+			case '\'':
+				if (escaped) {
```

Comment:
This needs to handle multi-line strings `"""` as well.

Reply:
Do you have any example?

Reply:
`extends Node2D

var s: String ="""
Reference
"""`

Reply:
AFAIK multiline `String`s are not supported in Godot 4.